frost-core/src/frost/identifier.rs

```diff
@@ -14,183 +14,87 @@ use crate::{Ciphersuite, Error, Field, Group, Scalar};
 /// over, corresponding to some x-coordinate for a polynomial f(x) = y.  MUST NOT be zero in the
 /// field, as f(0) = the shared secret.
 #[derive(Copy, Clone)]
-pub struct Identifier<C: Ciphersuite>(pub(crate) Scalar<C>);
+pub struct Identifier<C>(u16, PhantomData<C>);
 
-impl<C> AsRef<Scalar<C>> for Identifier<C>
+impl<C> Identifier<C>
 where
     C: Ciphersuite,
 {
-    fn as_ref(&self) -> &Scalar<C> {
-        &self.0
+    // Convert the identifier to a Scalar.
```

Comment:
```suggestion
    // Convert the identifier to a Scalar.
    //
    // Ideally this would be a From<Identifier<C>> for Scalar<C> impl, but rustc
    // doesn't like that
```